config/class-sync.php

```diff
@@ -63,7 +83,65 @@ public function filter_cron_schedules( $schedule ) {
 		return $schedule;
 	}
 
-	public function do_cron() {
+	public function get_blogs_to_sync() {
+		return $this->blogs_to_sync;
+	}
+
+	public function queue_sync_for_blog() {
+		// Queue the sync only from admin or CLI, to keep it out of frontend request path
+		if ( ! is_admin() && ! Context::is_wp_cli() ) {
+			return;
+		}
+		$blog_id = get_current_blog_id();
+
+		// To avoid performance issues, don't add if the count would surpass BLOGS_TO_SYNC_LIMIT.
+		// Can rely on the default cron schedules to sync the rest.
+		if ( count( $this->blogs_to_sync ) >= self::BLOGS_TO_SYNC_LIMIT ) {
```

Comment:
We will also limit the number of blogs we sync via a cron trigger, to limit inadvertently triggering a mass update on SDS that might cause problems.

We will rely on our cronjobs for when there is a mass update.